utilities/transactions/write_unprepared_transaction_test.cc

```diff
@@ -272,53 +346,81 @@ TEST_P(WriteUnpreparedTransactionTest, RecoveryTest) {
   }
 }
 
-TEST_P(WriteUnpreparedTransactionTest, UnpreparedBatch) {
+// Test whether logs containing unprepared/prepared batches are kept even
+// after memtable finishes flushing, and whether they are removed when
+// transaction commits/aborts.
+//
+// Similar to TransactionTest/TwoPhaseLogRollingTest.
+TEST_P(WriteUnpreparedTransactionTest, MarkLogWithPrepSection) {
   WriteOptions write_options;
   TransactionOptions txn_options;
-  // This causes writes into DB for every marker.
+  // batch_size of 1 causes writes to DB for every marker.
   txn_options.max_write_batch_size = 1;
   const int kNumKeys = 10;
 
+  WriteOptions wopts;
+  wopts.sync = true;
+
   for (bool prepare : {false, true}) {
     for (bool commit : {false, true}) {
       ReOpen();
-      Transaction* txn = db->BeginTransaction(write_options, txn_options);
-      WriteUnpreparedTxn* wup_txn = dynamic_cast<WriteUnpreparedTxn*>(txn);
-      txn->SetName("xid");
+      auto wup_db = dynamic_cast<WriteUnpreparedTxnDB*>(db);
+      auto db_impl = wup_db->db_impl_;
+
+      Transaction* txn1 = db->BeginTransaction(write_options, txn_options);
+      ASSERT_OK(txn1->SetName("xid1"));
+
+      Transaction* txn2 = db->BeginTransaction(write_options, txn_options);
+      ASSERT_OK(txn2->SetName("xid2"));
 
+      // Spread this transaction across multiple log files.
       for (int i = 0; i < kNumKeys; i++) {
-        txn->Put("k" + ToString(i), "v" + ToString(i));
-        ASSERT_EQ(wup_txn->GetUnpreparedSequenceNumbers().size(), i + 1);
+        ASSERT_OK(txn1->Put("k1" + ToString(i), "v" + ToString(i)));
+        if (i >= kNumKeys / 2) {
+          ASSERT_OK(txn2->Put("k2" + ToString(i), "v" + ToString(i)));
+        }
+
+        if (i > 0) {
+          db_impl->TEST_SwitchWAL();
+        }
       }
 
+      ASSERT_GT(txn1->GetLogNumber(), 0);
```

Comment:
why > 0?

Reply:
I'm just checking that the log number on both transactions are set, because we have written unprepared batches.